m4opt/_cli/schedule.py

```diff
@@ -275,7 +281,17 @@ def schedule(
         observer_locations = mission.observer_location(obstimes)
 
     with status("evaluating field of regard"):
-        target_coords = mission.skygrid
+        if not isinstance(mission.skygrid, dict):
+            target_coords = mission.skygrid
+        elif skygrid in mission.skygrid:
+            target_coords = mission.skygrid[skygrid]
+        else:
+            typer.secho(
+                f"skygrid '{skygrid}' not found. Options: {', '.join(map(str, mission.skygrid.keys()))}",
+                fg=typer.colors.RED,
+            )
+            raise typer.Exit(1)
```

Comment:
```suggestion
            raise UsageError(
                f"skygrid '{skygrid}' not found. Options: {', '.join(map(str, mission.skygrid.keys()))}"
            )
```

And add `from click import UsageError`.

For uncaught exceptions that are instances of any of the [Click exception classes](https://click.palletsprojects.com/en/stable/api/#exceptions), typer will nicely format and print the exception message but hide the traceback.